components/ButtonBar/tests/unit/ButtonBarObservationTests.swift

```diff
@@ -0,0 +1,167 @@
+    item.isEnabled = false
+
+    do {
+      let enabled = buttonBar.subviews.flatMap { $0 as? MDCButton }.map { $0.isEnabled }
+      XCTAssertEqual(enabled, [item.isEnabled])
+    }
+  }
+
+  func testImageChangesAreObserved() {
+    let image1 = createImage(colored: .red)
+
+    let item = UIBarButtonItem(image: image1, style: .plain, target: nil, action: nil)
+    buttonBar.items = [item]
+    buttonBar.layoutSubviews()
+
+    do {
+      let images = buttonBar.subviews.flatMap { $0 as? MDCButton }.map { $0.image(for: .normal) }
+      XCTAssertEqual(images, [item.image])
+    }
+
+    // Change the value post-assignment
+    item.image = createImage(colored: .blue)
+
+    do {
+      let images = buttonBar.subviews.flatMap { $0 as? MDCButton }.map { $0.image(for: .normal) }
+      XCTAssertEqual(images, [item.image])
+    }
+  }
+
+  func testTagChangesAreObserved() {
+    let item = UIBarButtonItem(title: "Title", style: .plain, target: nil, action: nil)
+    item.tag = 100
+    buttonBar.items = [item]
+    buttonBar.layoutSubviews()
+
+    do {
+      let tags = buttonBar.subviews.flatMap { $0 as? MDCButton }.map { $0.tag }
+      XCTAssertEqual(tags, [item.tag])
+    }
+
+    // Change the value post-assignment
+    item.tag = 50
+
+    do {
+      let tags = buttonBar.subviews.flatMap { $0 as? MDCButton }.map { $0.tag }
+      XCTAssertEqual(tags, [item.tag])
+    }
+  }
+
+  func testTintColorChangesAreObserved() {
+    let item = UIBarButtonItem(title: "Title", style: .plain, target: nil, action: nil)
+    item.tintColor = .blue
+    buttonBar.items = [item]
+    buttonBar.layoutSubviews()
+
+    do {
+      let tintColors = buttonBar.subviews.flatMap { $0 as? MDCButton }.flatMap { $0.tintColor }
+      XCTAssertEqual(tintColors, [item.tintColor!])
+    }
+
+    // Change the value post-assignment
+    item.tintColor = .red
+
+    do {
+      let tintColors = buttonBar.subviews.flatMap { $0 as? MDCButton }.flatMap { $0.tintColor }
+      XCTAssertEqual(tintColors, [item.tintColor!])
+    }
+
+    // Verify that the tint color reverts to the default
+    item.tintColor = nil
+
+    do {
+      let tintColors = buttonBar.subviews.flatMap { $0 as? MDCButton }.flatMap { $0.tintColor }
+      XCTAssertEqual(tintColors, [buttonBar.tintColor])
+    }
+  }
+
+  func testTitleChangesAreObserved() {
+    let item = UIBarButtonItem(title: "LEFT", style: .plain, target: nil, action: nil)
+    buttonBar.items = [item]
+    buttonBar.layoutSubviews()
+
+    do {
+      let titles = buttonBar.subviews.flatMap { $0 as? MDCButton }.map { $0.title(for: .normal) }
+      XCTAssertEqual(titles, [item.title])
+    }
+
+    // Change the value post-assignment
+    item.title = "NEW TITLE"
+
+    do {
+      let titles = buttonBar.subviews.flatMap { $0 as? MDCButton }.map { $0.title(for: .normal) }
+      XCTAssertEqual(titles, [item.title])
+    }
+  }
+
+  private func createImage(colored color: UIColor) -> UIImage {
```

Comment:
Super-nit: Prefer putting this method at the top so maintainers reading the file can see that the method exists locally (vs. globally) before it is used?

Reply:
Done.